core/src/main/java/jenkins/install/SetupWizard.java

```diff
@@ -0,0 +1,124 @@
+    
+    public SetupWizard(File _root, ServletContext context) throws IOException, InterruptedException, ReactorException {
+        super(_root, context);
+        this.context = context;
+        generateToken();
+    }
+
+    /**
+     * Skip much of the initialization during setup
+     * @throws IOException
+     * @throws InterruptedException
+     * @throws ReactorException
+     */
+    @Override
+    protected void init() throws IOException, InterruptedException, ReactorException {
+        // do bare minimum here
+        // doing this early allows InitStrategy to set environment upfront
+        final InitStrategy is = new SetupWizardInitStrategy();
+
+        // initialization consists of ...
+        executeReactor( is,
+            // load and prepare plugins... this should only be included js-module plugins at initial install time
+            pluginManager.initTasks(is)
+        );
+    }
+    
+    public void doSetupComplete(Jenkins j) {
+        context.setAttribute(WebAppMain.APP, j);
+    }
+
+    private void generateToken() {
+        randomUUID = UUID.randomUUID().toString();
+        LOGGER.info("SetupWizard SecurityToken: " + randomUUID);
+//        File lastExecVersion = InstallUtil.getLastExecVersionFile();
+//        FileUtils.write(lastExecVersion, randomUUID);
+    }
+    
+    public Object getAssets() {
+        return assetManager;
+    }
+    
+    public Object getI18n() {
+        return i18n;
+    }
+    
+    @Override
+    public Object getDynamic(String token) {
+        LOGGER.info("get dynamic: " + token);
+        if (!isAllowedSetupPage(token)) {
+            return this;
+        }
+        Object o = super.getDynamic(token);
+        return o;
+    }
+    
+    private boolean isAllowedSetupPage(String token) {
+        return "securityRealm".equals(token)
+            || "configureSecurity".equals(token);
+    }
+
+    @Override
+    public SecurityRealm getSecurityRealm() {
+        return super.getSecurityRealm();
+    }
+    
+    public GlobalSecurityConfigurationInclude getConfigureSecurity() {
+        if(configureSecurity == null) { // 'lazy loading' this is fine
+            configureSecurity = new GlobalSecurityConfigurationInclude();
+        }
+        return configureSecurity;
+    }
+    
+    public void complete() {
+        try {
+            cleanUp();
+            Jenkins j = new Hudson(root, context);
+            applySecuritySettings(j);
+            doSetupComplete(j); // set j as the active jenkins instance
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Apply any security settings from this Jenkins instance to the provided one
+     * @param j
+     */
+    public void applySecuritySettings(Jenkins j) {
+        j.setSecurityRealm(getSecurityRealm());
+        j.setAuthorizationStrategy(getAuthorizationStrategy());
+        j.setCrumbIssuer(getCrumbIssuer());
+    }
+}
```

Comment:
The `SetupWizard` class smells strange to me. The thinking behind this class needs to be clearly stated and verified.